scripts/emulate_app.sh

```diff
@@ -249,46 +243,53 @@ else
 
     echo -e "\nWaiting for AVD to boot..."
     wait_for $timeout_seconds check_boot
-    echo -e "\nWaiting for AVD to be online..."
     wait_for $timeout_seconds check_device_online
     echo $(adb -s emulator-5554 emu avd name | head -1)
     echo "Device online" && sleep 1
 
-    echo -e "\nInstalling APKs..."
-    i=0
-    step_complete=false
-    until [[ $step_complete == true ]]; do
-        if adb -s emulator-5554 install -r -t -d --abi "${abi}" \
-                "app/build/outputs/apk/debug/app-${abi}-debug.apk" &> "${test_report_dir}/apk_installation.txt"; then
-            step_complete=true
-            echo "Successfully installed APKs"
-        fi              
-        if [[ $i -ge 100 ]]; then
-            echo "Error: Failed to install APKs" >&2
-            echo "For more information see 'android/${test_report_dir}/apk_installation.txt'" >&2
-            exit 1
-        fi
-        i=$((i+1))
-        sleep 1
-    done
+    if "${boot_app}"; then
+        echo -e "\nRunning yarn install..."
+        yarn install
+
+        echo -e "\nBuilding APKs..."
+        ./gradlew assembleDebug -Psplitapk=true
 
-    # Store emulator info and start logging
-    adb -s emulator-5554 shell getprop &> "${test_report_dir}/getprop.txt"
-    adb -s emulator-5554 shell cat /proc/meminfo &> "${test_report_dir}/meminfo.txt"
-    adb -s emulator-5554 shell cat /proc/cpuinfo &> "${test_report_dir}/cpuinfo.txt"
-    adb -s emulator-5554 shell logcat -v threadtime -b main &> "${test_report_dir}/logcat.txt" &
+        echo -e "\nInstalling APKs..."
+        i=0
+        step_complete=false
+        until [[ $step_complete == true ]]; do
+            if adb -s emulator-5554 install -r -t -d --abi "${abi}" \
+                    "app/build/outputs/apk/debug/app-${abi}-debug.apk" &> "${test_report_dir}/apk_installation.txt"; then
+                step_complete=true
+                echo "Successfully installed APKs"
+            fi              
+            if [[ $i -ge 100 ]]; then
+                echo "Error: Failed to install APKs" >&2
+                echo "For more information see 'android/${test_report_dir}/apk_installation.txt'" >&2
+                exit 1
+            fi
+            i=$((i+1))
+            sleep 1
+        done
+
+        # Store emulator info and start logging
```

Comment:
you probably want to store the emulator info and logs even if you are not booting app?